webserver/__init__.py

```diff
@@ -131,6 +134,17 @@ def before_request_gdpr_check():
         elif current_user.is_authenticated and current_user.gdpr_agreed is None:
             return redirect(url_for('index.gdpr_notice', next=request.full_path))
 
+    @app.before_request
+    def prod_https_login_redirect():
+        """ Redirect to HTTPS on the production login page
+        """
+        split_url = urlparse.urlsplit(request.url)
+        if split_url.scheme == 'http' \
+            and split_url.netloc == PRODUCTION_BASE_URL \
+            and split_url.path == url_for('login.index'):
+            redirect_url = urlparse.urljoin(PRODUCTION_BASE_URL_HTTPS, url_for('login.index', next=request.args.get('next')))
```

Comment:
It seems a bit funny to do this before _every_ request, but only check if the url path is the login page. Why not do only when rendering the login button?

I was going to suggest that we could do this redirect on every url if it’s https, but that would cause problems with non-GET requests, so perhaps we should skip that.